src/org/exist/storage/BrokerPools.java

```
@@ -0,0 +1,319 @@
+     * Returns whether or not a database instance is configured.
+     *
+     * @param instanceName The name of the database instance
+     * @return <code>true</code> if it is configured
+     */
+    public static boolean isConfigured(final String instanceName) {
+        final Lock readLock = instancesLock.readLock();
+        readLock.lock();
+        try {
+            final BrokerPool instance = instances.get(instanceName);
+            if (instance == null) {
+                return false;
+            } else {
+                return instance.isInstanceConfigured();
+            }
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    /**
+     * Returns the broker pool for the default database instance (if it is configured).
+     *
+     * @return The broker pool of the default database instance
+     *
+     * @throws EXistException If the instance is not available (not created, stopped or not configured)
+     */
+    public static BrokerPool getInstance() throws EXistException {
+        return getInstance(DEFAULT_INSTANCE_NAME);
+    }
+
+    /**
+     * Returns a broker pool for a database instance.
+     *
+     * @param instanceName The name of the database instance
+     * @return The broker pool
+     *
+     * @throws EXistException If the instance is not available (not created, stopped or not configured)
+     */
+    public static BrokerPool getInstance(final String instanceName) throws EXistException {
+        //Check if there is a database instance with the same id
+        final Lock readLock = instancesLock.readLock();
+        readLock.lock();
+        try {
+            final BrokerPool instance = instances.get(instanceName);
+            if (instance != null) {
+                //TODO : call isConfigured(id) and throw an EXistException if relevant ?
+                return instance;
+            } else {
+                throw new EXistException("Database instance '" + instanceName + "' is not available");
```

I don't know where you find case 1. `instancesInitializtionException` get into game only when `instances` have nothing for given name.

With current code you hide actual exception from end-user by "is not available" exception (that gives nothing and force to ask user to check logs)